Service discovery (#1258)
kyma-project/busola: docs/features.md

title: Feature flags
---

> The document explains the usage of feature flags in Busola, as well as lists and describes all the feature flags that are available in Kyma Dashboard and provides their configuration examples:

Reviewer: ```suggestion
The document explains the usage of feature flags in Busola, lists and describes all the available feature flags, and provides their configuration examples:
```
Maybe we can shorten the intro like above?

#### Features priority

> Features in Busola initialisation is based on `stage` property, which can take one of the values listed here:

Reviewer: ```suggestion
Initialisation of the Busola features is based on the `stage` property, which can take one of the following values:
```

> - `PRIMARY` - the feature will be resolved while the app bootstraps. Features that should be immediately visible should be set as `PRIMARY` (e.g. main navigation structure).

Reviewer: ```suggestion
- `PRIMARY` - the feature is resolved while the application bootstraps. Features that should be immediately visible must be set as `PRIMARY` (for example, main navigation structure).
```

> - `SECONDARY` - the feature will be resolved after the app is ready, it should be used for non-critical features (e.g. additional navigation nodes).

Reviewer: ```suggestion
- `SECONDARY` - the feature is resolved after the application is ready, it must be used for non-critical features (for example, additional navigation nodes).
```

> - \<other value> - the feature is loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.

Reviewer: ```suggestion
- `{other_value} - the feature is loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.
```

> - \<other value> - the feature will be loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.

Reviewer: ```suggestion
- \<other value> - the feature is loaded only on-demand, most often by the iframe. Use the `useFeature` hook to request usage of such feature.
```

> Note that some features need to be run before the app starts the bootstrap process (e.g. SSO_LOGIN), so they are out of normal feature flow.

Reviewer: ```suggestion
Note that some features must be run before the application starts the bootstrap process (for example, SSO_LOGIN), so they are out of the normal feature flow.
```
